charts/newrelic-infrastructure-beta/templates/kubelet/daemonset.yaml

```diff
@@ -0,0 +1,275 @@
+{{- if (.Values.kubelet.enabled) }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.labels" . | nindent 4 }}
+  name: {{ template "newrelic.fullname" . }}-kubelet
+  {{- with .Values.kubelet.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- with .Values.updateStrategy }}
+  updateStrategy: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "newrelic.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: kubelet
+  template:
+    metadata:
+      annotations:
+        checksum/nri-kubernetes: {{ include (print $.Template.BasePath "/kubelet/configmap.yaml") . | sha256sum }}
+        {{- if not (include "newrelic.licenseCustomSecretName" .) }}
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.common.agentConfig }}
+        checksum/agent-config: {{ include (print $.Template.BasePath "/agent-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.integrations_config }}
+        checksum/integrations_config: {{ include (print $.Template.BasePath "/kubelet/integrations-configmap.yaml") . | sha256sum }}
+        {{- end }}
```

Comment:
The agent supports hot reloading so _in theory_ this should not be necessary. However we know this sometimes behaves erratically on k8s. Should we force a restart of the agent pod in this case? @newrelic/coreint @newrelic/caos

Reply:
Inotify can be unreliable so let's keep it for now.